src/main/java/com/labsynch/labseer/domain/Lot.java

```diff
@@ -1701,4 +1677,36 @@ public static Collection<Lot> fromJsonArrayToLots(String json) {
     public String toString() {
         return ReflectionToStringBuilder.toString(this, ToStringStyle.SHORT_PREFIX_STYLE);
     }
+
+    public static HashMap<Long, String> getOriginallyDrawnAsStructuresByParentIds(List<Long> parentIds) {
+
+        Query q = entityManager().createNativeQuery("WITH AS_DRAWN_STRUCTS as ("
+            + "select p.id as parent_id, coalesce(l.as_drawn_struct, p.mol_structure) as_drawn_struct, ROW_NUMBER () OVER (PARTITION BY p.id order by l.lot_number asc) as rn "
+            + "from parent p "
+            + "left join salt_form s "
+            + "on p.id = s.parent "
+            + "left join lot l on s.id = l.salt_form and (l.ignore is null or l.ignore = :ignore) and l.as_drawn_struct is not null "
+            + "where p.id in (:parent_ids)) "
+            + "select parent_id, as_drawn_struct from AS_DRAWN_STRUCTS where rn = 1")
+        .setParameter("parent_ids", parentIds)
+        .setParameter("ignore", false);
+
+        // Convert the result list into a hashmap of parent id to mol structure
+        HashMap<Long, String> parentIdsToMolStructures = new HashMap<Long, String>();
+        List<Object[]> results = q.getResultList();
+        for (Object[] result : results) {
+            // Cast BigInteger value to Long
+            Long parentId = ((BigInteger) result[0]).longValue();
+            parentIdsToMolStructures.put(parentId, (String) result[1]);
+        }
+
+        // Verify that each of the original pIdGroup parent ids exist in the result set
```

Comment:
`nit-pick`: I think even though this method is only called in context of methods that have the `pIdGroup` naming, we could probably not reference it here to avoid confusions for people looking at this method for the first time? :thinking:

Reply:
Thanks, missed that when I refactored this into it's own method